planner/core/planbuilder.go

```
@@ -1030,7 +1031,46 @@ func checkHintedSQL(sql, charset, collation, db string) error {
 	return nil
 }
 
+func (b *PlanBuilder) buildCreateBindPlanFromPlanDigest(v *ast.CreateBindingStmt) (Plan, error) {
+	bindableStmt := stmtsummary.StmtSummaryByDigestMap.GetBindableStmtByPlanDigest(v.PlanDigest)
+	if bindableStmt == nil {
+		return nil, errors.New("sql digest '" + v.PlanDigest + "' invalid")
+	}
+
+	parser4binding := parser.New()
+	originNode, err := parser4binding.ParseOneStmt(bindableStmt.Query, bindableStmt.Charset, bindableStmt.Collation)
+	if err != nil {
+		logutil.BgLogger().Debug("[sql-bind] parse origin SQL failed in create binding from history",
+			zap.String("SQL", bindableStmt.Query), zap.Error(err))
+	}
+	bindSQL := bindinfo.GenerateBindSQL(context.TODO(), originNode, bindableStmt.PlanHint, true, bindableStmt.Schema)
+	var hintNode ast.StmtNode
+	hintNode, err = parser4binding.ParseOneStmt(bindSQL, bindableStmt.Charset, bindableStmt.Collation)
+	if err != nil {
+		logutil.BgLogger().Debug("[sql-bind] parse bind SQL failed in create binding from history",
```

ditto. Better to return this error directly.